AsciiPixel/src/AsciiPixel.jl

```diff
@@ -0,0 +1,39 @@
+module AsciiPixel
+
+using ImageBase: restrict
+using ImageCore
+using Crayons
+
+export ascii_encode
+
+include("colorant2ansi.jl")
+include("ascii_encode.jl")
+
+const colormode = Ref{TermColorDepth}(TermColor8bit())
+
+"""
+    set_colordepth(bit::Int)
+
+Sets the terminal color depth to the given argument.
+"""
+function set_colordepth(bit::Int)
+    if bit == 8
+        colormode[] = TermColor8bit()
+    elseif bit == 24
+        colormode[] = TermColor24bit()
+    else
+        error("Setting color depth to $bit-bit is not supported, valid mode are:
+          - 8bit (256 colors)
+          - 24bit")
+    end
+    colormode[]
+end
+
+set_colordepth(bit::AbstractString) = set_colordepth(parse(Int, replace(bit, r"[^0-9]"=>"")))
+
+function __init__()
+    # use 24bit if the terminal supports it
+    lowercase(get(ENV, "COLORTERM", "")) in ("24bit", "truecolor") && set_colordepth(24)
```

Comment:
We can add a `is_24bit_supported() = lowercase(get(ENV, "COLORTERM", "")) in ("24bit", "truecolor")` util and use it here; this allows downstream packages to reuse the codes instead of copying it.

The reason is that checking `COLORTERM` doesn't always give the correct result; it's very likely to change in the future. E.g., For Windows we might want to unconditionally set `is_24bit_supported() = false` for Julia < 1.6. By providing a function we can quickly propagate the changes to downstream packages.